centreon/www/widgets/servicegroup-monitoring/src/index.php

```diff
@@ -120,12 +120,37 @@
 
 
 $orderby = "name ASC";
-if (isset($preferences['order_by']) && trim($preferences['order_by']) != "") {
-    $orderby = $preferences['order_by'];
+
+$allowedOrderColumns = ['name'];
+
+const ORDER_DIRECTION_ASC = 'ASC';
+const ORDER_DIRECTION_DESC = 'DESC';
+const DEFAULT_ENTRIES_PER_PAGE= 10;
+
+$allowedDirections = [ORDER_DIRECTION_ASC, ORDER_DIRECTION_DESC];
+$defaultDirection = ORDER_DIRECTION_ASC;
+
+$orderByToAnalyse = isset($preferences['order_by'])
+    ? trim($preferences['order_by'])
+    : null;
+
+if ($orderByToAnalyse !== null) {
+    [$column, $direction] = explode(' ', $orderByToAnalyse);
+    $direction = $direction !== null ? strtoupper($direction) : $defaultDirection;
```

Comment:
If the direction is not defined in the $orderByToAnalysis variable, an error will be generated because key 1 does not exist.

Reply:
fixed